### PR TITLE
Rename Perl 6 to Raku

### PR DESCRIPTION
**Pull request by xfix**
## Description

Perl 6 has changed its name to Raku.

This PR introduces new extensions that aren't really used much (most code is using `.p6` or `.pm6`) considering the support for those was introduced a week ago. However, according to https://github.com/perl6/problem-solving/blob/master/solutions/language/Path-to-Raku.md, `.raku`, `.rakutest` and `.rakumod` are the new extensions, and support for those is already in main Raku implementation (see changelog https://github.com/rakudo/rakudo/releases/tag/2019.11, although worth noting this only adds `.rakumod` as other extensions don't require implementation change). I don't think introducing those extensions will be an issue as they unambiguously refer to Raku and shouldn't conflict with anything else.

## Checklist:

- [ ] **I am associating a language with a new file extension.**
  - [ ] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3AFOOBAR+KEYWORDS+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  
  - [ ] I have included a syntax highlighting grammar: https://github-lightshow.herokuapp.com/
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

- [ ] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  
  - Old: https://github-lightshow.herokuapp.com/
  - New: https://github-lightshow.herokuapp.com/

- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

- [ ] **I am changing the color associated with a language**
  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -4352,6 +4328,36 @@ Ragel:
...
+  - ".p6"
+  - ".p6l"
+  - ".p6m"
+  - ".pl"
+  - ".pl6"
+  - ".pm"
+  - ".pm6"
+  - ".raku"
+  - ".rakumod"
+  - ".rakutest"
```
Comment (Alhadis) [inline lib/linguist/languages.yml]: I'm afraid you'll need to remove these extensions for now, as we only support extensions once they've seen [sizeable usage](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-an-extension-to-a-language) on GitHub.com. [Harvester](https://github.com/Alhadis/Harvester) counted only a negligible number of each:

* [`.raku`](https://github.com/search?p=1&q=extension%3Araku+NOT+nothack&type=Code):  71 files, 25 repos, 22 users
* [`.rakumod`](https://github.com/search?q=extension%3Arakumod+NOT+nothack&type=Code): 169 files, 6 repos, 5 users
* [`.rakutest`](https://github.com/search?q=extension%3Arakutest+NOT+nothack&type=Code): 7 files, 3 repos, 3 users

We can always add them later. See #4219.

Comment (xfix) [reply lib/linguist/languages.yml]: Fair enough. Can those extensions be tracked separately in #4219? Those extensions are new (but are intended to become the main extensions), so I wouldn't expect them to be particularly popular quite yet (that said, they are much more popular than when I checked them).

Comment (Alhadis) [reply lib/linguist/languages.yml]: Certainly. I'll ping @pchaigno to update the list, as he's the one responsible for monitoring extension usage over time.

**Comment by xfix** [comment]
I think this can be merged without adding extensions if that's the only blocker?

**Comment by Alhadis** [comment]
@xfix We don't merge unless a GitHub staff member has given their approval; that's just policy. 😉 In any case, changes won't take effect across the site until the next release of Linguist has been made, which generally happens roughly once a month.

**Comment by xfix** [comment]
Fair enough, I wasn't sure with "Pending Popularity" label.

**Comment by pchaigno** [comment]
Sorry, I should have explained. I'm adding that label to remember to collect usage for these extensions. This can be merged as soon as @lildude gives his :+1:

**Pull request merged**